Codec/Archive/Tar/Write.hs

```diff
@@ -39,6 +43,16 @@ import qualified "os-string" System.OsString.Posix as PS
 write :: [Entry] -> LBS.ByteString
 write es = LBS.concat $ map putEntry es ++ [LBS.replicate (512*2) 0]
 
+write' :: [GenEntry OsPath TarPath LinkTarget] -> IO LBS.ByteString
+write' es = interleavedByteStringConcat $ map putEntry' es ++ [pure $ LBS.replicate (512*2) 0]
+
+interleavedByteStringConcat :: [IO LBS.ByteString] -> IO LBS.ByteString
+interleavedByteStringConcat [] = return LBS.empty
```

Comment:
Could it be expressed via `foldr`, similar to `Codec.Archive.Tar.Pack.interleavedSequence`?

Reply:
I don't know. I find `foldr` version of `interleavedSequence` quite hard to grasp. The `(f . ) . g)` is IMO point-free taken too far.